providers/NMSReflectionProvider/src/net/ess3/nms/refl/providers/ReflServerStateProvider.java

```diff
@@ -0,0 +1,47 @@
+package net.ess3.nms.refl.providers;
+
+import net.ess3.nms.refl.ReflUtil;
+import net.ess3.provider.ServerStateProvider;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.logging.Logger;
+
+public class ReflServerStateProvider implements ServerStateProvider {
+    private final Object nmsServer;
+    private final MethodHandle nmsIsRunning;
+
+    public ReflServerStateProvider(Logger logger) {
+        Object serverObject = null;
+        MethodHandle isRunning = null;
+        Class<?> nmsClass = ReflUtil.getNMSClass("MinecraftServer");
+        try {
+            serverObject = nmsClass.getMethod("getServer").invoke(null);
+            isRunning = MethodHandles.lookup().findVirtual(nmsClass, "isRunning", MethodType.methodType(boolean.class));
+        } catch (Exception e) {
+            logger.severe("Something's gone terribly wrong!");
+            e.printStackTrace();
+            logger.severe("Something's gone terribly wrong!");
```

Comment:
Bit verbose, we probably just need the stack trace

```suggestion
            e.printStackTrace();
```